### PR TITLE
Add options for ConfigFactory to handle customConfig properly

### PR DESCRIPTION
**Pull request by mkondratek**
Fixes https://linear.app/sourcegraph/issue/CODY-4264/a-comment-in-cody-settingsjson-breaks-cody-initialization-stuck-at.

## Test plan
1. Have the cody_settings.json like this:
```
{
  // other providers...
  "openctx.providers": {
    "https://gist.githubusercontent.com/thenamankumar/3708f084fb2abd57adafe7f14620bdf7/raw/e7c7059cb5b04f6feead002e7b3a90c5b1a4bb96/provider.js": true
  }
}
```
2. Cody should initialize

### CODE REVIEW

File: src/main/kotlin/com/sourcegraph/config/ConfigUtil.kt [hunk 1]
```diff
@@ -141,11 +139,11 @@ object ConfigUtil {
 
     return try {
       val text = customConfigContent ?: getSettingsFile(project).readText()
-      val config = ConfigFactory.parseString(text).resolve()
+      val config = JsonParser.parseString(text).asJsonObject
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/config/ConfigUtil.kt]: ConfigFactory does not render json. The rendered config contained the comment with # instead of //. Using JsonParser fixes the problem.

Comment (pkukielka) [reply src/main/kotlin/com/sourcegraph/config/ConfigUtil.kt]: Other option would be adding `.setComments(false)` in the current code.
I thought Lightened Config was a bit more forgiving when parsing e.g. additional commas.
Maybe we can write simple test for that and check how those corner cases looks in case of both approaches?

Comment (mkondratek) [reply src/main/kotlin/com/sourcegraph/config/ConfigUtil.kt]: fixed

File: src/main/kotlin/com/sourcegraph/config/ConfigUtil.kt [hunk 2]
```diff
@@ -141,11 +141,13 @@ object ConfigUtil {
 
     return try {
       val text = customConfigContent ?: getSettingsFile(project).readText()
-      val config = ConfigFactory.parseString(text).resolve()
+      var config = ConfigFactory.parseString(text).resolve()
       additionalProperties.forEach { (key, value) ->
-        config.withValue(key, ConfigValueFactory.fromAnyRef(value))
```
Comment (mkondratek) [inline src/main/kotlin/com/sourcegraph/config/ConfigUtil.kt:146]: There was a bug 😄 configs are immutable

File: src/test/kotlin/com/sourcegraph/config/ConfigUtilTest.kt [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+    assertEquals(
+        "indentation-based",
+        parsed
+            .get("cody")
+            .asJsonObject
+            .get("experimental")
+            .asJsonObject
+            .get("foldingRanges")
+            .asString)
```
Comment (mkondratek) [inline src/test/kotlin/com/sourcegraph/config/ConfigUtilTest.kt:33]: `withValue` adds the field as 
```
"cody": {
  "experimental": {
    "foldingRanges": "indentation-based"
  }
}
```
instead of 
```
"cody.experimental.foldingRanges": "indentation-based"
```

Is it a problem? 🤔

File: src/main/kotlin/com/sourcegraph/config/ConfigUtil.kt (hunk 2 above)
Comment (mkondratek) [reply src/main/kotlin/com/sourcegraph/config/ConfigUtil.kt:146]: Basically, we ignored "foldingRanges": "indentation-based" - how important for us is this option?

File: src/test/kotlin/com/sourcegraph/config/ConfigUtilTest.kt (hunk 3 above)
Comment (pkukielka) [reply src/test/kotlin/com/sourcegraph/config/ConfigUtilTest.kt:33]: IMHO It is not, those are equivalent in VSC config.

**Review by pkukielka — APPROVED** [review]
LGTM
